app/javascript/app/phone-internationalization.js

```diff
@@ -10,38 +10,97 @@ const selectedInternationCodeOption = () => {
-    '[data-international-phone-form] .otp_delivery_preference_voice',
-  ));
-  const smsRadio = /** @type {HTMLInputElement?} */ (document.querySelector(
-    '[data-international-phone-form] .otp_delivery_preference_sms',
-  ));
+/**
+ * @return {HTMLInputElement[]}
+ */
+const getOTPDeliveryMethods = () =>
+  Array.from(document.querySelectorAll('.js-otp-delivery-preference'));
+
+/**
+ * Returns true if the delivery option is valid for the selected option, or false otherwise.
+ *
+ * @param {string} delivery
+ * @return {boolean}
+ */
+const isDeliveryOptionSupported = (delivery) =>
+  selectedInternationCodeOption().getAttribute(`data-supports-${delivery}`) !== 'false';
 
-  if (!(phoneRadio && smsRadio)) {
-    return;
+/**
+ * @param {string} delivery
+ * @return {string=}
+ */
+function getHintTextForDisabledDeliveryOption(delivery) {
+  // i18n-tasks-use t('two_factor_authentication.otp_delivery_preference.voice_unsupported')
+  // i18n-tasks-use t('two_factor_authentication.otp_delivery_preference.sms_unsupported')
+  let hintText = I18n.t(
+    `two_factor_authentication.otp_delivery_preference.${delivery}_unsupported`,
+  );
+
+  if (hintText) {
+    const location = selectedInternationCodeOption().dataset.countryName;
+    hintText = hintText.replace('%{location}', location);
   }
 
-  const deliveryMethodHint = /** @type {HTMLElement} */ (document.querySelector(
-    '#otp_delivery_preference_instruction',
-  ));
-  const selectedOption = selectedInternationCodeOption();
-
-  const supportsSms = selectedOption.dataset.supportsSms === 'true';
-  const supportsVoice = selectedOption.dataset.supportsVoice === 'true';
-
-  smsRadio.disabled = !supportsSms;
-  phoneRadio.disabled = !supportsVoice;
-
-  if (supportsVoice) {
-    deliveryMethodHint.innerText = I18n.t(
-      'two_factor_authentication.otp_delivery_preference.instruction',
-    );
-  } else {
-    smsRadio.click();
-    deliveryMethodHint.innerText = I18n.t(
-      'two_factor_authentication.otp_delivery_preference.phone_unsupported',
-    ).replace('%{location}', selectedOption.dataset.countryName);
+  return hintText;
+}
+
+/**
+ * @param {string=} hintText
+ */
+function setHintText(
+  hintText = I18n.t('two_factor_authentication.otp_delivery_preference.instruction'),
+) {
+  const hintElement = document.querySelector('#otp_delivery_preference_instruction');
+  if (hintElement) {
+    hintElement.textContent = hintText;
+  }
+}
+
+/**
+ * Returns true if all inputs are disabled, or false otherwise.
+ *
+ * @param {HTMLInputElement[]} inputs
+ * @return {boolean}
+ */
+const isAllDisabled = (inputs) => inputs.every((input) => input.disabled);
+
+/**
+ * Returns the next non-disabled input in the set of inputs, if one exists.
+ *
+ * @param {HTMLInputElement[]} inputs
+ * @param {number} index
+ * @return {HTMLInputElement=}
+ */
+const getNextEnabledInput = (inputs, index) =>
+  [...inputs.slice(index + 1), ...inputs.slice(0, index)].find((input) => !input.disabled);
```

Comment:
the two slices were a little tricky for me to follow, I feel like we could maybe do something similar with a modulus?

```suggestion
const getNextEnabledInput = (inputs, startIndex) => {
  for (let idxOffset = 1, len = inputs.length; idxOffset < len; ++idxOffset) {
    const nextInput = inputs[(index + idxOffset) % len];
    if (!nextInput.disabled) {
      return nextInput;
    }
  }
}
```

Or maybe this is not easier at all.

Reply:
There's also only two options? So maybe we just unroll this and do something that could be like a ternary?

Reply:
One of the goals of the refactoring was to make it a little less hard-coded to the specific two options, since I think it contributed to some of the previous inconsistency. But I agree the slice dance here could be a bit confusing.

An alternative that could work just as well could be to change it to something where we find the first input which isn't disabled and not care about its position relative to the "current" one:

```suggestion
const getFirstEnabledInput = (inputs) => inputs.find((input) => !input.disabled);
```

This would only differ in a scenario where we'd have more than two options, e.g.

```
[ 1 (Enabled) ] [ 2 (Disabled) ] [ 3 (Enabled) ]
getNextEnabledInput(inputs, 1)
// Before: 3
// After: 1
```

I think it's fair to say it's a bit premature to anticipate needing that, so I can simplify 👍

Reply:
Updated in 31d4c2a.

Reply:
big fan of the simplification 👍